lib/Optimizer/Transforms/StatePreparation.cpp

```diff
@@ -35,6 +35,11 @@ namespace cudaq::opt {
 
 using namespace mlir;
 
+namespace cudaq {
+std::vector<std::complex<double>>
+readGlobalConstantArray(cudaq::cc::GlobalOp &global);
```

Comment:
Let's put the declaration in a header file. I believe this one belongs in Factory.h, which is where the other constant extracting functions are.

Reply:
Addressed in 3e71dee865f261ba1ba9379e172eb4b68ccab2d7